From fd7fc491486294d1f62d52ddd1ba4161e870db71 Mon Sep 17 00:00:00 2001
From: Erickson Silva de Oliveira <Erickson.SilvadeOliveira@windriver.com>
Date: Tue, 14 Oct 2025 16:40:26 -0300
Subject: [PATCH] Add label platform/application to pods

Add support to 'app.starlingx.io/component' to be overwritten
by user override, with possible values being 'platform' and
'application'. With 'platform' being the default value. This
change will also restart the pods if the label in namespace
change.

Signed-off-by: Erickson Silva de Oliveira <Erickson.SilvadeOliveira@windriver.com>
---
 templates/deployment.yaml | 2 ++
 values.yaml               | 1 +
 2 files changed, 3 insertions(+)

diff --git a/templates/deployment.yaml b/templates/deployment.yaml
index d7535b8..6d9b6e3 100644
--- a/templates/deployment.yaml
+++ b/templates/deployment.yaml
@@ -24,12 +24,14 @@ spec:
   template:
     metadata:
       labels:
+        app.starlingx.io/component: {{ ternary "application" "platform" .Values.isApplicationLabel}}
         {{- include "metrics-server.selectorLabels" . | nindent 8 }}
       {{- with .Values.podLabels }}
         {{- toYaml . | nindent 8 }}
       {{- end }}
       {{- with .Values.podAnnotations }}
       annotations:
+        configchecksum: {{ toYaml .Values.isApplicationLabel | sha256sum | trunc 63 }}
         {{- toYaml . | nindent 8 }}
       {{- end }}
     spec:
diff --git a/values.yaml b/values.yaml
index 18c3be3..fe11cea 100644
--- a/values.yaml
+++ b/values.yaml
@@ -46,6 +46,7 @@ apiService:
   # The PEM encoded CA bundle for TLS verification
   caBundle: ""

+isApplicationLabel: false
 commonLabels: {}
 podLabels: {}
 podAnnotations: {}
--
2.34.1
